v3/tasks/20250221-v3-cow-swap-fee-burner/input.ts

```diff
@@ -0,0 +1,22 @@
+import { Task, TaskMode } from '@src';
+
+export type CowSwapFeeBurnerDeployment = {
+  Vault: string;
+  AppData: string;
+  ComposableCow: string;
+  CowVaultRelayer: string;
+};
+
+const Vault = new Task('20241204-v3-vault', TaskMode.READ_ONLY);
+
+//TODO: Set our own app data before deployment. https://explorer.cow.fi/appdata?tab=encode
```

Comment:
Can we just do this now? What are we waiting on here? Looks like we need to come up with a JSON and encode it on that site.

Reply:
I'm not sure who needs to approve this data. Maybe Maxis or someone else?

Reply:
looks mostly like metadata of the burner/order config? happy to review if needed :)

Reply:
@gosuto-inzasheru  I'd say just tell us what you want the metadata to be (maybe referencing whatever it is for the Vyper original version), and we'll generate the app data from that. Then we can replace the TODO comment with:

// Appdata generated from the following metadata: <whatever it is>.
// See https://explorer.cow.fi/appdata?tab=encode.

Reply:
imo something like this?:
```
{
  "appCode": "CowSwapFeeBurner.sol-b38851d",
  "metadata": {
    "orderClass": {
      "orderClass": "limit"
    },
    "hooks": {
      "version": "0.1.0"
    },
    "signer": "0x123"
  },
  "version": "1.3.0"
}
```
notes:
- `appCode`: im just trying to point to the version of the deployed contract. not sure if that commit hash is final?
- `orderClass` should be limit imo given we give a min amount out to buy, right?
- `signer` should be address of the smart contract that is signing the order eip1271 style

there is also the `slippageBips`, which im not sure if we are implementing in any way? i guess not really, since we are dealing with a limit order, so can be left out

Reply:
I don't think we need appCode here - that sort of thing is usually handled with versioning on the contract. So maybe we should add versioning to the burner contract. The standard implementation lets you call `version()` on the contract, and it returns a JSON like:
{"name":"WeightedPool","version":1,"deployment":"20241205-v3-weighted-pool"}

See https://etherscan.io/address/0x527d0E14acc53FB040DeBeae1cAb973D23FB3568#readContract, last function.

Reply:
hmm afaik the watch towers are simply either listening to all events or you can whitelist a set of addresses to watch.

this whole metadata blob is mostly for us: to be able to track down an order back to how it was placed. like for example here: https://explorer.cow.fi/orders/0xd5c9171243f0c572f10ef4e22d564edc73aab242ac2eb79c7589fa753d9a20ca9ff471f9f98f42e5151c7855fd1b5aa906b1af7e67895ea0. appdata here simply exposes the fact that the order was placed through the safe app, and some additional 'meta' parameters

the use case i see is that if we would have multiple cow burners or something in place at some point (or different versions), then we could determine which one was used to place a specific order in case we need to debug